### ISSUE TITLE
Fix `selector-no-qualifying-type` false positives for `:has()`

### ISSUE THREAD

**Issue by stof** [stylelint/stylelint#9103]
### What minimal example or steps are needed to reproduce the bug?

```css
.form-field:has(input:disabled) {
    color: gray;
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "selector-no-qualifying-type": true
  }
}
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIB0AzCAnAtgWhQSxgBsp4ALAQwGcAKPAOwAcBXAF3ij0vICNCYoAlAAJgAHTpDJQyIQzwhAc3TkAngG5xAXxAAaEPj4A5cpjiIYADxMM+SMJUq7wEOvgUIQlhhhZDYKciZCXzEJIXQgmEp5UKkhAHJKIhgwFgxsOghsAEcmckI8FBV6BWwWFQYYePkWCJhxSU0dLQ06J0hXPAUAMQxMchYPSnK+AroWOxc3JEwAK0c9WAZHRFihURBhlVH6Fg35DcIBqL3dBvXNkaJd7A63bGHyOihydCh9i6OWE42tcRBNEA)

### Which Stylelint-related dependencies are you using?

```json
{
    "stylelint": "^17.3.0",
    "stylelint-config-standard-scss": "^17.0.0",
}
```

### What did you expect to happen?

No error reported. `:has(input)` is not qualifying the current element. It applies a condition on *child* nodes.

### What actually happened?

This problem is reported:

```
1:17-22
    error
    Unexpected qualifying type selector "input" in ".form-field:has(input:disabled)" (selector-no-qualifying-type)
```

**Comment by jeddy3** [comment]
@stof Thanks for the report and for using the template.

I can [reproduce the issue](https://stylelint.io/demo/#N4Igxg9gJgpiBcID0AqABJATpmYAuANgJ5oB2EeaAZgQIYDm9MUaKSAOqbfABYQBuMTGmABfTgEtSABwCueeFAkBnWgCMCzEeNKdUGCNlyESNBkxa1laGAEdZE-mjwQ0tAHRUIrsQANWHKSe3vAA7jxCMAAUtACU2pzBEPAqMfFinHroUlg4+MTUdIxabIleyTxWaQlB5fDkeNUZurUhlcoxvAJC6TpJ9RRRUnIKSqoazL2cIAA0IFQSmgBytAC2cIgwAB5r0pruYMrKs+AQpAv0CCDb0oaUsFS0sgSUwJxoaJjPMMrwIu8fNDsEDKGCafCGAC05Eh9loBAkVCIUnokLwRGkMGBfzwXyxpA+OlEAG5pnNIOcJPQAGKGVa0PBXZTozQI0h4A5nC7uVYAK2Oc1g0mOiDeBKBIJZYKkeGxEroeB+stmAOBzKIrJlkIpF0hzNopCgtEwUDlwIVSuBnB0IFEQA) with simple selectors in `:has()` and `:not()`:

```css
/* correctly not flagged */
a:hover {}
input:disabled {}

/* correctly flagged as equiv to a.foo {} */
.foo:where(a) {}
.foo:is(a) {}

/* incorrectly flagged */
.foo:has(a) {}
.foo:not(a) {}

.foo:has(a:hover) {}
.foo:not(input:disabled) {}
```

We'll want to ignore selectors inside the non-grouping functional pseudo-classes of `:has()` and `:not()`.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/contributor-guide/rules#fix-a-bug-in-a-rule) in the Contributor guide.

**Comment by stof** [comment]
`:not` could be legitimately considered as qualifying the element.

**Comment by jeddy3** [comment]
Let's limit the change to `:has()`.

**Comment by romainmenke** [comment]
Makes sense to limit to `:has()` in this case.

In the `groupByCompoundSelectors` function I intended to produce all compound selectors for a given selector.

So `.foo:is(.bar)` produces `.foo.bar`.

But `:has()` is different from `is`, `where` and `not` in that the rightmost part of whatever exists inside `:has()` doesn't refer to the same element as the compound on the outside.
